### DISCUSSION TITLE
Posible to configure pnpm to have node_modules per package but no .pnpm folder?

### DISCUSSION THREAD

**Discussion by mikecfisher** [pnpm/pnpm#5739] [Q&A]
So I'm building an Expo React Native app in a pnpm managed monorepo. The problem is React Native by default doesn't work well with symlinks. Expo recommends using Yarn Workspaces but we really want to use pnpm. I've heard the solution to use React Native and pnpm is to turn on `node-linker=hoisted` which flattens all the `node_modules` at the root of the repo. While this does "work` it also just lumps all the dependencies of the whole monorepo into a sock drawer. 

What I would like to do is have each app in my `apps/` folder and each package in my `packages/` folder have their own `node_modules` folder. While it seems this is the default behavior for pnpm it also seems that everything is symlinked through a `.pnpm` folder which breaks React Native. 

Can anyone give me a suggestion on how to make this work?

**Comment by weyert** [comment]
Did you try https://pnpm.io/npmrc#symlink ? I am wondering if setting this setting to `false` improves things for you
This seems to refer to symlinks, PNPM and React Native might be helpful: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3575

**Reply by weyert** [reply]
https://www.npmjs.com/package/@rnx-kit/metro-resolver-symlinks

**Reply by mikecfisher** [reply]
Thanks! I'm actually already using that rnx-kit library and while it helps in development it gets a little wonky when I try to do builds with expo application services. 

First when I run a build it gives this message 

```
It looks like that you are using a custom metro.config.js that does not extend @expo/metro-config.
This can result in unexpected and hard to debug issues, like missing assets in the production bundle.
We recommend you to abort, fix the metro.config.js, and try again.
Learn more on customizing Metro
✔ Would you like to abort? … no
```

Then unless I have `node-linker=hoisted` enabled the builds completely fail.

**Comment by mikecfisher** [comment]
Also I tried `symlink=false` but then the `node_modules` folders are completely empty so that wont work sadly.
